src/asInput/index.jsx

```diff
@@ -129,6 +129,7 @@ const asInput = (WrappedComponent, inputType = undefined, labelFirst = true) =>
 
     getLabel() {
       return (
+        // eslint-disable-next-line jsx-a11y/label-has-for
```

Comment:
I'm seeing a [`jsx-a11y/label-has-for`](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/label-has-for.md) violation here

![image](https://user-images.githubusercontent.com/8136030/37670052-b3ecea70-2c3e-11e8-86be-9ee3720f405e.png)

@fysheets you were the last person to touch this - do you have any idea why this accessibility violation is occurring and what the solution might be?

Reply:
Looking at the documentation that you linked, for components example is using a nested `Label` component where the ID value is passed explicitly through. Our code is using `{this.state.id}` which I'm going to assume the linter isn't smart enough to match up to the ID being set/passed through for the input.